### PR TITLE
Dropping the none probabilities

### PR DESCRIPTION
**Pull request by mojtaba-komeili**
**Patch description**
Running some evaluations that were using the OPT model as their backend server, I noticed that sometime they were crashing while calculating the PPL. Digging deeper it turned out there is a `None` value in the `token_logprobs` returned from the server. Adding this filter resolved the crashing issue.

Screen Shot 2022-11-30 at 12 47 51 PM

**PS**: I am not sure if this has any unwanted consequences on the validity of the metric.

### CODE REVIEW

**Comment by klshuster** [comment]
uhh cc @stephenroller why would we be getting `None` back?

**Comment by stephenroller** [comment]
the first token doesn't have a probability associated with it. we did this to be compatible with openai

File: projects/bb3/agents/utils.py [hunk 1]
```diff
@@ -650,7 +650,9 @@ def compute_perplexities(
             start_label = [i for i, off in enumerate(text_off) if off <= prompt_len]
             assert len(start_label) > 0
             start_label = start_label[-1]
-            all_log_probs = result['choices'][0]['logprobs']['token_logprobs']
+            all_log_probs = [
+                lp for lp in result['choices'][0]['logprobs']['token_logprobs'] if lp
```
Comment (klshuster) [inline projects/bb3/agents/utils.py:654]: could we change to `if lp is not None`, as `lp==0` could happen

Comment (mojtaba-komeili) [reply projects/bb3/agents/utils.py]: It shouldn't be an issue in practice given it comes from the softmax. But who know what we will be using in the future, so added it lol.

**Review by klshuster — APPROVED** [review]
thanks for the patch!

**Pull request merged**